### PR TITLE
product implementation first time

### PR DESCRIPTION
**Pull request by AliHashem20**
...

### CODE REVIEW

**Comment by AliHashem20** [comment]
what was done :
Implemented product (add, get) endpoints with it schema
Test the database connection
Install and configure eslint package
How to test :
test it using postman in case there is no user interface 
links: 1- get all products ( http://localhost:4000/product )
         2- get single product by its id (http://localhost:4000/product/647b8142b83188ef50750e66)
         3- add product (post method) http://localhost:4000/product/add

File: app.js [hunk 1]
```diff
@@ -7,17 +7,15 @@ const app = express();
...
+    })
+    .catch((err) => {
     });
-  })
-  .catch((err) => {
-    console.log(err);
-  });
 
-app.use("/product", productRoute);
+app.use('/product', productRoute);
```
Comment (imadsaad12) [inline app.js:21]: In RestAPIs the URLs have some constraints you can check it here: https://restfulapi.net/rest-api-design-tutorial-with-example/
so it should be /products (with s)

File: controllers/productController.js [hunk 2]
```diff
@@ -1,44 +1,39 @@
-const Product = require("../models/productModel");
-const serviceProduct = require("../models/productModel");
+const serviceProduct = require('../services/productService');
```
Comment (imadsaad12) [inline controllers/productController.js:1]: lets use destruction instead of access function using serviceProduct . const {addProduct}=require('../services/productService')

File: controllers/productController.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const addproduct = async (req, res) => {
+  const data = req.body;
+  try {
+    await serviceProduct.addProduct(data);
+    return res.status(200).send('Product added successfully');
```
Comment (imadsaad12) [inline controllers/productController.js:7]: No need for return since we are at the end of function

File: controllers/productController.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      return res.json(products);
+    }
+  } catch (error) {
+    return res.status(400);
+  }
+};
+
+const getProduct = async (req, res) => {
+  try {
+    const id = req.params.id;
```
Comment (imadsaad12) [inline controllers/productController.js:28]: let use destructure : const { id }= req.params

File: routes/productRoute.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+// eslint-disable-next-line new-cap
+const route = express.Router();
+const productController = require('../controllers/productController');
+
+route.post('/add', productController.addproduct);
```
Comment (imadsaad12) [inline routes/productRoute.js:6]: As RestAPIs official documentation (https://restfulapi.net/rest-api-design-tutorial-with-example/ ) all URLs should look the same for DELETE,GET,POST or PUT so the URL for adding product should be ( / ) not ( /product )

File: routes/productRoute.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+// eslint-disable-next-line new-cap
+const route = express.Router();
+const productController = require('../controllers/productController');
```
Comment (imadsaad12) [inline routes/productRoute.js:4]: lets use destructure here

File: services/productService.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+const Product = require('../models/productModel');
+
+const addProduct = async (data) => {
+  const product = new Product(data);
+  await product.save();
```
Comment (imadsaad12) [inline services/productService.js:5]: To add product we can use await Product.create() instead of these 2 lines

File: services/productService.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+const allProducts = async () => {
+  const products = await Product.find();
+  return products;
```
Comment (imadsaad12) [inline services/productService.js:10]: in javascript when we function that contains onlu online we can omit return line and asyn await keywords so we can do it like this :
 const allProducts=()=>Product.find()

File: package.json [hunk 9]
```diff
@@ -4,13 +4,21 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "lint": "eslint ."
```
Comment (imadsaad12) [inline package.json:8]: lets change script to : eslint --fix . 
this will automatically fix most issues when we call it , for example it will replace all double quote with single quote

File: models/productModel.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose');
+
+const productSchema = new mongoose.Schema(
```
Comment (imadsaad12) [inline models/productModel.js:3]: it should be productsSchema

File: models/productModel.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+      },
+      image: {
+        type: String,
+        required: true,
+      },
+    },
+    {timestamps: true},
+);
+
+const Product = mongoose.model('product', productSchema);
```
Comment (imadsaad12) [inline models/productModel.js:33]: it should be Products

**Review by imadsaad12 — CHANGES_REQUESTED** [review]
All functions are working perfectly , I left you minor comments to be addressed when you fix these comments let me now. 
Great work !!

File: services/productService.js [hunk 12]
```diff
@@ -1,22 +1,15 @@
-const Product = require('../models/productModel');
+const Products = require('../models/productModel');
 
-const addProduct = async (data) => {
-  const product = new Product(data);
-  await product.save();
+const createProduct = async (data) => {
+  await Products.create(data);
```
Comment (imadsaad12) [inline services/productService.js:4]: you can also  remove async/await we will do it in another PR

**Review by imadsaad12 — APPROVED** [review]
Good job ALi , you are good to go
